packages/medusa/src/interfaces/payment-processor.ts

```diff
@@ -0,0 +1,212 @@
+import {
+  Address,
+  Customer,
+  PaymentSessionStatus,
+  ShippingMethod,
+} from "../models"
+import { MedusaContainer } from "../types/global"
+
+export type PaymentProcessorContext = {
+  shipping_address?: Address | null
+  shipping_methods?: ShippingMethod[]
+  email: string
+  currency_code: string
+  amount: number
+  resource_id?: string
+  customer?: Customer
+}
+
+export type PaymentProcessorSessionResponse = {
+  update_requests: { customer_metadata: Record<string, unknown> }
+  session_data: Record<string, unknown>
+}
+
+export interface PaymentProcessorError {
+  error: string
+  code: number
+  details: any
+}
+
+/**
+ * The new payment service plugin interface
+ */
+export interface PaymentProcessor {
+  /**
+   * Return a unique identifier to retrieve the payment plugin provider
+   */
+  getIdentifier(): string
+
+  /**
+   * Initiate a payment session with the external provider
+   */
+  initiatePayment(
```

Comment:
thought: Sorry, I actually liked init for the SDK/client/etc better than initializePayment, so maybe we could stick with that and then rename this one to startPayment. Wdyt? 😅

Reply:
InitiatePayment was previously createPayment. I ve removed the init method. Do you want me to re add the init? I am not sure i ve followed

Reply:
Sorry, I misplaced my comment. To avoid blocking this PR further, I'm fine with just using `init`. My apologies for going back and forth on this one.